stellargraph/core/graph_standard.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The standardised StellarGraph implementation.
+
+"""
+__all__ = ["StandardStellarGraph"]
+
+from typing import Iterable, Any, Mapping, Optional, Union
+import itertools
+
+from .graph import StellarGraph
+from ..data.node_data import NodeData
+from ..data.edge_data import EdgeData
+from ..data.edge_cache import EdgeCache
+from .schema import GraphSchema, EdgeType
+
+
+class StandardStellarGraph(StellarGraph):
+    """
+    Implementation based on encapsulating user-supplied edge and node data.
+    """
+
+    def __init__(self, is_directed=False, edges=None, nodes=None, **kwargs):
+        self._is_directed = is_directed
+        self._node_data = nd = self._get_node_data(nodes, kwargs)
+        self._edge_data = ed = self._get_edge_data(edges, kwargs)
+        self._edge_cache = EdgeCache(kwargs.get("edge_cache"), ed, nd)
```

Comment:
I thought about passing in the edge cache type, but then this opens up the possibility of a user passing in their own implementation. At the time, I had specific reservations about why this could be bad, but I can't remember what they were (it might have been about exposing the interface or about testing, not sure).